### PR TITLE
Adding Redux State Management and UI Components for Rockets, Missions, and Dragons

### PR DESCRIPTION
**Pull request by ahmidou09**
**Pull Request Description:**
This pull request implements the required features and functionalities as outlined in the project requirements.

**General Requirements:**
- All linter errors have been resolved.
- Gitflow has been followed for the project's version control.
- Documentation has been updated in a professional manner.

**HTML/CSS & JavaScript Requirements:**
- Best practices for HTML, CSS, and JavaScript have been adhered to.

**Project Requirements: Config & Basic Setup:**
- Created a React App.
- Installed React Redux and React Router.
- App logo has been added using a free image.
- Routes and view components for rockets, missions, and profile have been created.
- Navigation links are implemented using `<NavLink />` to indicate active sections.
- Separate directories have been created for Redux state slices: rockets, missions, dragons.

**Redux: Fetch Data and Update Redux Store:**
- Fetched data from SpaceX API endpoints for rockets, missions, and dragons.
- Dispatched actions to store the fetched data in the Redux store.
- Redux store now contains data structures for rockets, missions, and dragons.

**Render UI: Lists:**
- Used `useSelector()` Redux Hook to select state slices for rockets and missions.
- Rendered lists of rockets and missions in their respective routes.
- Styled the application using React Bootstrap components.

**Rendering Data:**
- Rendered a list of rockets using the first image in the array of `flickr_images`.
- Rendered a table with missions' data.
- Rendered a list of dragons using the first image in the array of `flickr_images`.

**Redux: Actions and Reducers for Booking Rockets/Dragons and Joining Missions:**
- Implemented actions and reducers for booking rockets and dragons, as well as joining missions.
- Dispatched actions to update the store with reserved status for selected rockets, dragons, and missions.

**Redux: Actions and Reducers for Canceling Rockets/Dragons and Leaving Missions:**
- Implemented actions and reducers for canceling reservations of rockets and dragons, as well as leaving missions.
- Dispatched actions to update the store with updated reservation status for rockets, dragons, and missions.

**Render UI: Conditional Component Rendering:**
- Conditionally rendered "Reserved" badge and "Cancel reservation" button for rockets and dragons.
- Conditionally rendered "Active Member" badge and "Leave Mission" button for joined missions.

**Render UI: My Profile Section:**
- Implemented a two/three column layout for the profile section.
- Listed rockets, dragons, and missions reserved or joined by the user using `filter()`.

**Technical Setup:**
- Set up the repository on GitHub following Gitflow.
- Set up React and Redux for state management.
- Configured testing libraries including React Testing Library and Jest.

### CODE REVIEW

File: src/__test__/Dragon.test.js
Comment (mukhammadrizooff) [inline src/__test__/Dragon.test.js:1]: - [ ] [REQUIRED] Good job so far! All `tests` shoud be passed but in this file has a problem. Please fix your failure file 🙏🏼 

image

File: README.md [hunk 1]
```diff
@@ -1,70 +1,272 @@
...
+    - [Prerequisites](#prerequisites)
+    - [Setup](#setup)
+    - [Install](#install)
+    - [Usage](#usage)
+  - [👥 Authors ](#-authors-)
+  - [🔭 Future Features ](#-future-features-)
+  - [🤝 Contributing ](#-contributing-)
+  - [⭐️ Show your support ](#️-show-your-support-)
+  - [🙏 Acknowledgments ](#-acknowledgments-)
+  - [📝 License ](#-license-)
```
Comment (mukhammadrizooff) [inline README.md]: - [ ] [OPTIONAL] It would be nice if you deploy your project and paste the link of deployment to your `README` 🙏🏼

**Review by mukhammadrizooff — CHANGES_REQUESTED** [review]
Hi @TEAM, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @Team, 


68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966


Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⚡ 

✅ Descriptive PR 👍 
✅ Nice design implementation 🔥 
✅ Great job fixing the requested changes 🔥 
✅ Good commit messages 👍 
✅ Linters are passing 👏 
✅ Tests are now passing 👏 



## Optional suggestions

- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**